src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs

```diff
@@ -190,5 +202,32 @@ private static HashAlgorithmType MapHashAlgorithmType(Interop.libssl.DataHashAlg
                     return HashAlgorithmType.None;
             }
         }
+        
+        private static int GetHashAlgorithmBitSize(Interop.libssl.DataHashAlgorithm mac)
+        {
+            switch (mac)
+            {
+                case Interop.libssl.DataHashAlgorithm.SSL_MD5:
+                    return MD5_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA1:
+                    return SHA1_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST94:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST89MAC:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA256:
+                    return SHA256_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA384:
+                    return SHA384_HashKeySize;
+
+                default:
+                    return 0;
```

Comment:
> is there a similar macro to indicate "modern libssl" so that #if can be inverted?

There aren't any macros/defines at all yet that I'm aware of, I was suggesting we create one named OPENSSL_0_9_8 to use on OSX.
If you wanted it inverted, you could use `#ifndef OPENSSL_0_9_8`.  I'm not sure how many versions of OpenSsl we are planning on supporting at one time. I would assume we are supporting 0.9.8 for use on OSX, and the last known good version for use on Linux. On my Ubuntu machine, it looks like I have 1.0.1.  @bartonjs - can you confirm?

> Btw I don't find "SSL_CIPHER_get_digest_nid " in 1.0.* existing versions. Which version is this present?

According to https://www.openssl.org/docs/manmaster/ssl/SSL_CIPHER_get_name.html, it exists in the latest, i.e. "master", version but not in 1.0.2 or 1.0.1.